src/placement/pl_map.c

```diff
@@ -615,15 +615,16 @@ pl_select_leader(daos_obj_id_t oid, uint32_t shard_idx, uint32_t grp_size,
 	 *      The one with the lowest f_seq will be elected as the leader
 	 *      to avoid leader switch.
 	 */
-	rdg_idx = shard_idx / replicas;
-	start = rdg_idx * replicas;
-	replica_idx = (oid.lo + rdg_idx) % replicas;
+	rdg_idx = shard_idx / grp_size;
+	start = rdg_idx * grp_size;
+	replica_idx = (oid.lo + rdg_idx) % grp_size;
 	preferred = start + replica_idx;
 
-	for (i = 0, off = preferred, pos = -1; i < replicas;
-	     i++, replica_idx = (replica_idx + 1) % replicas,
+	for (i = 0, off = preferred, pos = -1; i < grp_size;
+	     i++, replica_idx = (replica_idx + 1) % grp_size,
```

Comment:
here replicas should equal to grp_size, right?

Reply:
Actually, they are not equal. Since during OSA, the object will be added a few extra shards for writing to the old/new shard.  For example, for RP2_1G object,  if there is one shard(1) being migrated, then there will be 3 shards for the object.

shard_0
shard_1_old
shard_1_new

so grp_size is 3, but replicas is 2.